sound/soc/sof/pcm.c

```diff
@@ -419,13 +419,22 @@ static int sof_pcm_open(struct snd_pcm_substream *substream)
 				   SNDRV_PCM_HW_PARAM_PERIOD_SIZE,
 				   le32_to_cpu(caps->period_size_min));
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_DISABLE_NOWAKEUP)
+	/* set runtime config - remove no period wakeup */
+	runtime->hw.info = SNDRV_PCM_INFO_MMAP |
+			  SNDRV_PCM_INFO_MMAP_VALID |
+			  SNDRV_PCM_INFO_INTERLEAVED |
+			  SNDRV_PCM_INFO_PAUSE |
+			  SNDRV_PCM_INFO_RESUME;
+#else
 	/* set runtime config */
 	runtime->hw.info = SNDRV_PCM_INFO_MMAP |
 			  SNDRV_PCM_INFO_MMAP_VALID |
 			  SNDRV_PCM_INFO_INTERLEAVED |
 			  SNDRV_PCM_INFO_PAUSE |
 			  SNDRV_PCM_INFO_RESUME |
 			  SNDRV_PCM_INFO_NO_PERIOD_WAKEUP;
+#endif
```

Comment:
@fredoh9 this can be condensed a little bit. u can set runtime->hw_info to everything but SNDRV_PCM_INFO_NO_PERIOD_WAKEUP outside the if. 
And in the if you can do runtime->hw.info  |= SNDRV_PCM_INFO_NO_PERIOD_WAKEUP

Reply:
and the commit should probably be squashed

Reply:
@ranj063 Yes! thank you for good comment. Will do that.

Reply:
@fredoh9 
please remove it from outside if
#if !IS_ENABLE()
runtime->hw.info |= NO_PERIOD_WAKEUP
#endif

Reply:
Sorry, i missed this comment. I thought removing flag when the debugging feature is enabled also good idea. but setting and clearing in next line may cause confusing also. I agree. Will re-push after fixing this.